app/service/rest_svc.py

```diff
@@ -280,6 +281,14 @@ def _get_stopping_conditions(data):
         if new_stopping_conditions:
             return [{s.get('trait'): s.get('value')} for s in new_stopping_conditions]
 
+    def _gen_goals(self, goal_data):
+        goal = goal_data
+        goal['count'] = int(goal['count'])
+        if goal['count'] < 0:
+            goal['count'] = 2 ** 20
```

Comment:
Just a big number to prevent full goal set from being satisfied and thus ceasing operation?

Reply:
Yep. In theory, 2^20 is far more matches for any fact we will ever find, but if that changes, it's not hard to patch.

Reply:
Use the class attribute here (as mentioned above)

Reply:
Take care of.